crates/solver/src/settlement_submission.rs

```diff
@@ -155,6 +159,7 @@ impl SubmissionError {
             Self::Timeout => SettlementSubmissionOutcome::Timeout,
             Self::Revert => SettlementSubmissionOutcome::Revert,
             Self::Canceled => SettlementSubmissionOutcome::Cancel,
+            Self::Disabled(_) => SettlementSubmissionOutcome::Disabled,
             Self::Other(_) => SettlementSubmissionOutcome::SimulationRevert,
```

Comment:
Unrelated to this PR, but does it makes sense for this to be the `Failure` outcome that we used to have? From what I understand, you can get here by errors like failing to establish a TCP connection to the node when submitting a transaction which seems misclassified as a "simulation revert".

Not suggesting fixing it in this PR, just something I noticed while reviewing.

Reply:
I think you are right. Fixed.